### PR TITLE
mypy/stubgen: add types interfaces

### PR DESCRIPTION
**Pull request by igalic**
this speculative pr adds autogenerated stubs (thanks mypy!) to the
libzfs project.

This is a starting point for a discussion, if you so what.

### CODE REVIEW

File: libzfs.pyi [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+    state = ... # type: Any
+    def __init__(self, *args, **kwargs): ...
+    def __getstate__(self): ...
+
+class ZfsConverter:
+    __init__ = ... # type: Any
+    to_native = ... # type: Any
+    to_property = ... # type: Any
+
+class datetime(date):
```
Comment (igalic) [inline libzfs.pyi:297]:
```
error: Name 'date' is not defined
```

Comment (igalic) [reply libzfs.pyi:297]: this can be fixed with
```python
from datetime import date
```

**Review by igalic — COMMENTED** [review]
missing import

**Comment by araujobsd** [comment]
What is the purpose?

**Comment by igalic** [comment]
being able to [mypy](http://mypy.readthedocs.io/en/latest/) test a project that uses py-libzfs

**Comment by araujobsd** [comment]
It is an experimental project and very much intrusive, I would prefer see pytest covering libzfs rather than add an additional layer of complexity provided by an experimental project.

Maybe when it gets mature and other big projects adopt it we can consider, but for now, it is just a big NO.

Thanks for the PR anyway.

**Pull request closed without merging**